internal/beater/java_attacher/os_utils.go

```diff
@@ -0,0 +1,142 @@
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+//go:build !windows
+// +build !windows
+
+package javaattacher
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"os/user"
+	"path/filepath"
+	"strconv"
+	"syscall"
+)
+
+func (j *JavaAttacher) setRunAsUser(jvm *jvmDetails, cmd *exec.Cmd) error {
+	currentUser, err := user.Current()
+	if err != nil {
+		// If we cannot get the current user, then we assume apm-server
+		// is not the same as the target JVM user.
+		j.logger.Warnf("failed to get the current user: %v", err)
+	} else {
+		j.logger.Debugf("current user: %v", currentUser)
+	}
+
+	if currentUser.Gid != jvm.gid || currentUser.Uid != jvm.uid {
+		uid, gid, err := parseUserIds(jvm.uid, jvm.gid)
+		if err != nil {
+			return err
+		}
+		cmd.SysProcAttr = &syscall.SysProcAttr{}
+		cmd.SysProcAttr.Credential = &syscall.Credential{Uid: uint32(uid), Gid: uint32(gid)}
+	}
+	return nil
+}
+
+// getAttacherJar finds an attacher jar based on the given uid.
+// In POSIX-compliant systems, it would be an attacher jar owned by the given user and with 0600 access mode.
+// If such was not yet created, this function calls createAttacherTempDir to create it and keep a mapping for it
+func (j *JavaAttacher) getAttacherJar(uid string) string {
+	j.tmpAttacherLock.Lock()
+	defer j.tmpAttacherLock.Unlock()
+	attacherJar, uidMapped := j.uidToAttacherJar[uid]
+	if !uidMapped {
+		tmpAttacherJar, err := j.createAttacherTempDir(uid)
+		if err != nil {
+			j.logger.Errorf("failed to create tmp dir for user %v, using the bundled attacher jar", err)
+		} else {
+			attacherJar = tmpAttacherJar
+		}
+		j.uidToAttacherJar[uid] = attacherJar
+	}
+	return attacherJar
+}
+
+// createAttacherTempDir looks for a temp dir that is already mapped to the given user.
+// If such is not found, creates one as follows:
+// 	1.	create a temporary dir with 0700 access
+//	2.	copy the bundled attacher jar into the tmp dir with 0600 mode
+//	3.	change tmp dir and tmp attacher jar owner to the given user
+//	4.	keep a mapping of this jar to the user ID
+//
+// NOTE: this method is not thread-safe, so it should not be invoked concurrently by multiple goroutines
+func (j *JavaAttacher) createAttacherTempDir(uidS string) (string, error) {
+	uid, _, err := parseUserIds(uidS, "0")
+	if err != nil {
+		return "", err
+	}
+	bundledAttacherFile, err := os.Open(bundledJavaAttacher)
+	if err != nil {
+		return "", fmt.Errorf("failed to open bundled attacher jar: %w", err)
+	}
+	//goland:noinspection GoUnhandledErrorResult
```

Comment:
Do we need to commit IDE specific linting comments in the code?
```suggestion
```

Reply:
Probably not!
I am used to write to a repo where all main authors use the same IDE, where it does make sense